customcommands/web.go

```diff
@@ -725,3 +885,33 @@ func updateTemplateWithCountData(count int, templateData web.TemplateData, ctx c
 	}
 	templateData["AdditionalMessage"] = additionalMessage
 }
+
+func PublicCommandMW(inner http.Handler) http.Handler {
+	mw := func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		_, templateData := web.GetBaseCPContextData(ctx)
+		publicID := pat.Param(r, "cmd")
+		cc, err := models.CustomCommands(
+			models.CustomCommandWhere.PublicID.EQ(publicID)).OneG(r.Context())
+		if err != nil {
+			logger.Error(errors.WithStackIf(err))
+			http.Redirect(w, r, "/", http.StatusTemporaryRedirect)
+			return
+		}
+
+		if read, _ := web.IsAdminRequest(ctx, r); read {
+			http.Redirect(w, r, fmt.Sprintf("/manage/%d/customcommands/commands/%d/", cc.GuildID, cc.LocalID), http.StatusSeeOther)
```

Comment:
If someone comes from a public link to the CC, let them be on the public link whoever they are, the redirection here shouldn't happen if they admin or have read access to the CC.